package/MDAnalysis/analysis/nucleicacids.py

```diff
@@ -210,23 +253,150 @@ class WatsonCrickDist(NucPairDist):
     """
 
     def __init__(self, strand1: List[Residue], strand2: List[Residue],
-                 n1_name: str = 'N1', n3_name: str = "N3",
-                 g_name: str = 'G', a_name: str = 'A', u_name: str = 'U',
-                 t_name: str = 'T', c_name: str = 'C',
+                 nitrogen1_name: str = 'N1', nitrogen3_name: str = "N3",
+                 guanine_name: str = 'G', adenine_name: str = 'A', uracil_name: str = 'U',
+                 tymine_name: str = 'T', cytosine_name: str = 'C',
                  **kwargs) -> None:
-        sel1: List[mda.AtomGroup] = []
-        sel2: List[mda.AtomGroup] = []
-        strand = zip(strand1, strand2)
 
-        for s in strand:
-            if s[0].resname[0] in [c_name, t_name, u_name]:
-                a1, a2 = n3_name, n1_name
-            elif s[0].resname[0] in [a_name, g_name]:
-                a1, a2 = n1_name, n3_name
-            else:
-                raise ValueError(f"{s} are not valid nucleic acids")
+        selections: Tuple[List[mda.AtomGroup], List[mda.AtomGroup]] = self.select_strand_atoms(
```

Comment:
Instead of `selections` I would call it `strand_atomgroups` or something similar (maybe just `strands`), so that it's clear that it contains actual objects and not just "selection strings".